test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -211,6 +282,154 @@ public void CreateDisplayMetadata_DisplayAttribute_DescriptionFromResources()
             Assert.Equal("description from resources", context.DisplayMetadata.Description());
         }
 
+        // This is IMPORTANT. Product code needs to use GetPlaceholder() instead of .Placeholder. It's easy to regress.
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_PromptFromResources_WithLocalizer()
+        {
+            // Arrange
+            // Nothing on stringLocalizer should be called
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>();
+            stringLocalizerFactory
+                .Setup(s => s.Create(It.IsAny<Type>()))
+                .Returns(() => stringLocalizer.Object);
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory.Object);
+
+            var display = new DisplayAttribute()
+            {
+#if USE_REAL_RESOURCES
+                Description = nameof(Test.Resources.DisplayAttribute_Prompt),
+                ResourceType = typeof(Test.Resources),
+#else
+                Description = nameof(Test.Resources.DisplayAttribute_Prompt),
+                ResourceType = typeof(TestResources),
+#endif
+            };
+
+            var attributes = new Attribute[] { display };
+            var key = ModelMetadataIdentity.ForType(typeof(string));
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+
+            // Act
+            provider.CreateDisplayMetadata(context);
+
+            // Assert
+            Assert.Equal("prompt from resources", context.DisplayMetadata.Placeholder());
+        }
+
+        // This is IMPORTANT. Product code needs to use GetPlaceholder() instead of .Description. It's easy to regress.
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_PromptFromResources_NullLocalizer()
+        {
+            // Arrange
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory: null);
+
+            var display = new DisplayAttribute()
+            {
+#if USE_REAL_RESOURCES
+                Description = nameof(Test.Resources.DisplayAttribute_Prompt),
+                ResourceType = typeof(Test.Resources),
+#else
+                Description = nameof(Test.Resources.DisplayAttribute_Prompt),
+                ResourceType = typeof(TestResources),
+#endif
+            };
+
+            var attributes = new Attribute[] { display };
+            var key = ModelMetadataIdentity.ForType(typeof(string));
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+
+            // Act
+            provider.CreateDisplayMetadata(context);
+
+            // Assert
+            Assert.Equal("prompt from resources", context.DisplayMetadata.Placeholder());
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_LocalizerMissingProvidedKeys()
```

Comment:
I continue to object to the existence of this test. It confirms exactly the same thing as `CreateDisplayMetadata_DisplayAttribute_LocalizeProperties()` and no code change outside Moq could make one test fail and the other succeed.

Are you imagining MVC special-casing `Key == Value` or some such? Nothing like that exists.

Note you aren't even double-checking behaviour of code from the Localization repo. That's all mocked out.